crates/ty_python_semantic/resources/mdtest/liskov.md

```diff
@@ -0,0 +1,284 @@
+    with an argument of type `bool`, then the method `B.f` must also be callable with type `bool`
+    (though it is permitted for the override to also accept other types)
+1. Mutable attributes should only ever be overridden invariantly: if a mutable attribute `A.attr`
+    resolves to type `str`, it can only be overidden on a subclass with exactly the same type.
+
+## Method return types
+
+<!-- snapshot-diagnostics -->
+
+```pyi
+class Super:
+    def method(self) -> int: ...
+
+class Sub1(Super):
+    def method(self) -> int: ...  # fine
+
+class Sub2(Super):
+    def method(self) -> bool: ...  # fine: `bool` is a subtype of `int`
+
+class Sub3(Super):
+    def method(self) -> object: ...  # error: [invalid-method-override]
+
+class Sub4(Super):
+    def method(self) -> str: ...  # error: [invalid-method-override]
+```
+
+## Method parameters
+
+<!-- snapshot-diagnostics -->
+
+```pyi
+class Super:
+    def method(self, x: int, /): ...
+
+class Sub1(Super):
+    def method(self, x: int, /): ...  # fine
+
+class Sub2(Super):
+    def method(self, x: object, /): ...  # fine: `method` still accepts any argument of type `bool`
+
+class Sub4(Super):
+    def method(self, x: int | str, /): ...  # fine
+
+class Sub5(Super):
+    def method(self, x: int): ...  # fine: `x` can still be passed positionally
+
+class Sub6(Super):
+    # fine: `method()` can still be called with just a single argument
+    def method(self, x: int, *args): ...
+
+class Sub7(Super):
+    def method(self, x: int, **kwargs): ...  # fine
+
+class Sub8(Super):
+    def method(self, x: int, *args, **kwargs): ...  # fine
+
+class Sub9(Super):
+    def method(self, x: int, extra_positional_arg=42, /): ... # fine
+
+class Sub10(Super):
+    def method(self, x: int, extra_pos_or_kw_arg=42): ...  # fine
+
+class Sub11(Super):
+    def method(self, x: int, *, extra_kw_only_arg=42): ...  # fine
+
+class Sub12(Super):
+    # Some calls permitted by the superclass are now no longer allowed
+    # (the method can no longer be passed any arguments!)
+    def method(self, /): ...  # error: [invalid-method-override]
+
+class Sub13(Super):
+    # Some calls permitted by the superclass are now no longer allowed
+    # (the method can no longer be passed with exactly one argument!)
+    def method(self, x, y, /): ...  # error: [invalid-method-override]
+
+class Sub14(Super):
+    # Some calls permitted by the superclass are now no longer allowed
+    # (x can no longer be passed positionally!)
+    def method(self, /, *, x): ...  # error: [invalid-method-override]
+
+class Sub15(Super):
+    # Some calls permitted by the superclass are now no longer allowed
+    # (x can no longer be passed any integer -- it now requires a bool!)
+    def method(self, x: bool, /): ...  # error: [invalid-method-override]
+
+class Super2:
+    def method2(self, x): ...
+
+class Sub16(Super2):
+    def method2(self, x, /): ...  # error: [invalid-method-override]
```

Comment:
At this point it feels a bit arbitrary that we've stopped explaining the errors...